source/Ship.h

```diff
@@ -503,16 +505,19 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	bool isThrusting = false;
 	bool isReversing = false;
 	bool isSteering = false;
+	bool firePrimary = false;
 	double steeringDirection = 0.;
 	bool neverDisabled = false;
 	bool isCapturable = true;
 	bool isInvisible = false;
+	bool isPlayerFlagship = false;
 	int customSwizzle = -1;
 	double cloak = 0.;
 	double cloakDisruption = 0.;
 	// Cached values for figuring out when anti-missile is in range.
 	double antiMissileRange = 0.;
 	double weaponRadius = 0.;
+	double weaponRange = 0.;
```

Comment:
whats the difference between weapon range and weapon radius?

Reply:
`weaponRadius` describes how far from the center of the ship (i.e. 0,0) the furthest hardpoint is. `weaponRange` describes the maximum range amongst the weapons installed on the ship (excluding antimissile).

I use `weaponRange` to determine when a non player ship should enter the `Body::BodyState::FIRING` animation state. The logic here is, if the AI has an active enemy target selected, and that target is within some multiple of the maximum range of the ship's weapons, then the AI should play the firing sprite.